integr8ly/integreatly-operator
# MGDAPI-3419 possible fix and more logs to be collected

# Issue link
[MGDAPI-3419](https://issues.redhat.com/browse/MGDAPI-3419)

# What
check the comment in the JIRA 


# Verification steps
To fully verify you would need an installation with RHOAM pod on the cluster. Plus you would need to modify the code - right now the added portion of a code will be executed only when the error is encountered 5 times in a row. You could reuse the image I've been using: `quay.io/mvavilov/managed-api-service:mgdapi-3419`. It includes the following modifications:
1. The `FailureCounter` init value is set to 5
2. `connectToPostgres` and `restartPod` are called outside of an error catch statement on each reconcile loop. 

Verification:
This should be tested against a CCS cluster as the issue was spotted only on `useClusterStorage=false` installations. 
1. Install RHOAM by `USE_CLUSTER_STORAGE=false INSTALLATION_TYPE=managed-api IMAGE_FORMAT=<the image of your choice> DEV_QUOTA=10 make cluster/deploy` (to make the life easier, but the verification longer you might reach the `complete` stage using the `quay.io/integreatly/managed-api-service:master` and then flip the image) 
2. Open RHMI pod logs and Keycloak pods in the `rhsso` namespace
3. Watch for the Info message `Successfully connected` followed by the warning `Reached error counter limit of 5, Restarting redhat-rhoam-rhsso pods`
4. Keycloak pods should be deleted and recreated 
5. Before the info message you should see details of the Postgres instance - they should match the `rhsso-postgres-rhoam` in the redhat-rhoam-operator namespace.
